File: petcollector/backend/models/pet.py
from pydantic import BaseModel
from typing import Optional
import uuid
from datetime import datetime

class Pet(BaseModel):
    id: str = str(uuid.uuid4())
    name: str
    species: str
    description: str
    personality: str
    fun_fact: str
    rarity: str
    tags: list[str]
    photo_url: Optional[str] = None
    collected_by: str = "anonymous"
    collected_at: str = str(datetime.utcnow())

File: petcollector/frontend/src/types/pet.ts
export interface Pet {
  id: string
  name: string
  species: string
  description: string
  personality: string
  fun_fact: string
  rarity: 'Common' | 'Uncommon' | 'Rare' | 'Legendary'
  tags: string[]
  photo_url: string | null
  collected_by: string
  collected_at: string
}